File: MyBudget-front/src/app/models/transaction.ts
import { Account } from './account';

export class Transaction {
  id: number;
  description: string;
  amount: number;
  currency: string;
  account: Account;

  constructor(
    id: number,
    description: string,
    amount: number,
    currency: string,
    account: Account
  ) {
    this.id = id;
    this.description = description;
    this.amount = amount;
    this.currency = currency;
    this.account = account;
  }
}

File: MyBudget-back/src/main/java/com/example/mybudget/models/dto/Transaction.java
package com.example.mybudget.models.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Transaction {
    private Integer id;
    private String description;
    private BigDecimal amount;
    private String currency;
    private Account account;
}
